src/libos/crates/async-rt/src/executor.rs

```diff
@@ -70,82 +57,45 @@ impl Executor {
     }
 
     pub fn run_tasks(&self) {
-        let run_queue_id = self.next_run_queue_id.fetch_add(1, Ordering::Relaxed);
-        assert!(run_queue_id < self.parallelism);
-        let run_queue = &self.run_queues[run_queue_id as usize];
+        let thread_id = self.next_thread_id.fetch_add(1, Ordering::Relaxed) as usize;
+        assert!(thread_id < self.parallelism as usize);
+
         loop {
-            let task = {
-                let task_res = run_queue.try_recv();
-
-                if self.is_shutdown.load(Ordering::Relaxed) {
-                    return;
-                }
-
-                match task_res {
-                    Err(_) => {
-                        self.parks.park_timeout(
-                            run_queue_id as usize,
-                            core::time::Duration::from_millis(10),
-                        );
-                        continue;
-                    }
-                    Ok(task) => task,
-                }
-            };
-
-            let mut future_slot = task.future().lock();
-            let mut future = match future_slot.take() {
-                None => continue,
-                Some(future) => future,
-            };
-            drop(future_slot);
-
-            crate::task::current::set(task.clone());
-
-            let waker = waker_ref(&task);
-            let context = &mut Context::from_waker(&*waker);
-            if let Poll::Pending = future.as_mut().poll(context) {
-                let mut future_slot = task.future().lock();
-                *future_slot = Some(future);
+            let task_option = self.scheduler.get_runnable_task_for(thread_id);
+
+            if self.is_shutdown() {
```

Comment:
I think we should figure out what to do if there are still task in the run queues of the scheduler when the executor is shutdown. Adding a TODO would suffice for now.

Reply:
I agree